rfc3164/machine.go.rl

```diff
@@ -0,0 +1,204 @@
+package rfc3164
+
+import (
+    "fmt"
+    "time"
+
+    "github.com/influxdata/go-syslog/v2"
+	"github.com/influxdata/go-syslog/v2/common"
+)
+
+var (
+    errPrival         = "expecting a priority value in the range 1-191 or equal to 0 [col %d]"
+    errPri            = "expecting a priority value within angle brackets [col %d]"
+    errTimestamp      = "expecting a Stamp timestamp [col %d]"
+    errHostname       = "expecting an hostname (from 1 to max 255 US-ASCII characters) [col %d]"
+    errTag            = "expecting an alphanumeric tag (max 32 characters) [col %d]"
+    errContentStart   = "expecting a content part starting with a non-alphanumeric character [col %d]"
+    errContent        = "expecting a content part composed by visible characters only [col %d]"
+    errParse          = "parsing error [col %d]"
+)
+
+%%{
+machine rfc3164;
+
+include common "common.rl";
+
+# unsigned alphabet
+alphtype uint8;
+
+action mark {
+    m.pb = m.p
+}
+
+action set_prival {
+    output.priority = uint8(common.UnsafeUTF8DecimalCodePointsToInt(m.text()))
+	output.prioritySet = true
+}
+
+action set_timestamp {
+    if t, e := time.Parse(time.Stamp, string(m.text())); e != nil {
+        m.err = fmt.Errorf("%s [col %d]", e, m.p)
+        fhold;
+        fgoto fail;
+    } else {
+        output.timestamp = t
+        output.timestampSet = true
+    }
+}
+
+action set_hostname {
+    output.hostname = string(m.text())
+}
+
+action set_tag {
+    output.tag = string(m.text())
+}
+
+action set_content {
+    output.content = string(m.text())
+}
+
+action set_message {
+    output.message = string(m.text())
+}
+
+action err_prival {
+    m.err = fmt.Errorf(errPrival, m.p)
+	fhold;
+    fgoto fail;
+}
+
+action err_pri {
+    m.err = fmt.Errorf(errPri, m.p)
+	fhold;
+    fgoto fail;
+}
+
+action err_timestamp {
+    m.err = fmt.Errorf(errTimestamp, m.p)
+	fhold;
+    fgoto fail;
+}
+
+action err_hostname {
+	m.err = fmt.Errorf(errHostname, m.p)
```

Comment:
Does it matter to have a mix of tab and spaces?

Reply:
Gotta format it manually. Same old story 🤦‍♂ 😆

Reply:
Done